core/src/main/java/com/google/googlejavaformat/java/Formatter.java

```diff
@@ -203,6 +204,25 @@ public String formatSource(String input) throws FormatterException {
   }
 
   /**
+   * Format an input string (a Java compilation unit) and optimize imports into an output string.
+   *
+   * @param input the input string
+   * @param joi if not {@code null}, the imports are optimized accordingly
+   * @return the output string
+   * @throws FormatterException if the input string cannot be parsed
+   */
+  public String formatSource(String input, JavadocOnlyImports joi) throws FormatterException {
+    String output = formatSource(input, Collections.singleton(Range.closedOpen(0, input.length())));
+    if (joi != null) {
+      output = RemoveUnusedImports.removeUnusedImports(output, joi);
+      output = ImportOrderer.reorderImports(output);
+      // re-run format to remove empty lines and other artifacts introduced by imports mangling...
```

Comment:
Instead of

    package com.google.example;
     
    public class ExampleTest {}

it produces

    package com.google.example;
    
    
    
    
    
    
    public class ExampleTest {}